relay-server/src/metrics_extraction/sessions.rs

```diff
@@ -1,15 +1,173 @@
 use std::collections::BTreeMap;
+use std::fmt::{self, Display};
 
-use relay_common::{UnixTimestamp, Uuid};
+use relay_common::{MetricUnit, UnixTimestamp, Uuid};
 use relay_general::protocol::{
     AbnormalMechanism, SessionAttributes, SessionErrored, SessionLike, SessionStatus,
 };
-use relay_metrics::{Metric, MetricNamespace, MetricUnit, MetricValue};
+use relay_metrics::{
+    CounterType, Metric, MetricNamespace, MetricResourceIdentifier, MetricType, MetricValue,
+};
+
+/// Enumerates the metrics extracted from session payloads.
+#[derive(Clone, Debug, PartialEq)]
+pub enum SessionsMetric {
+    Session {
+        counter: CounterType,
+        timestamp: UnixTimestamp,
+        tags: SessionSessionTags,
+    },
+    User {
+        distinct_id: String,
+        timestamp: UnixTimestamp,
+        tags: SessionUserTags,
+    },
+    Error {
+        id: Uuid,
+        timestamp: UnixTimestamp,
```

Comment:
Random idea: we could get rid of the `timestamp` redundancy by having an `.into_metric(timestamp: UnixTimestamp)` method instead of `.into()`.

Reply:
i like it, I'll make a trait